<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Given an expression matrix from a bulk RNA-Seq experiment,
        pre-processes it and creates a shiny app for interactive data 
        analysis and visualisation. The app contains quality checks,
        differential expression analysis, volcano and cross plots,
        enrichment analysis and gene regulatory network inference,
        and can be customised to contain more panels by the user.">
<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/5/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Interactive Shiny App for Bulk Sequencing Data • bulkAnalyseR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Interactive Shiny App for Bulk Sequencing Data">
<meta property="og:description" content="Given an expression matrix from a bulk RNA-Seq experiment,
        pre-processes it and creates a shiny app for interactive data 
        analysis and visualisation. The app contains quality checks,
        differential expression analysis, volcano and cross plots,
        enrichment analysis and gene regulatory network inference,
        and can be customised to contain more panels by the user.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/5/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/5/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">
<!-- preferably CSS --><link rel="stylesheet" href="preferably.css">
<link id="css-code-light" rel="stylesheet" href="code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">
<script src="darkswitch.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">bulkAnalyseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/bulkAnalyseR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/IntegratingMultiomics.html">bulkAnalyseR: Integrating multi-omics or external data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Core-Bioinformatics/bulkAnalyseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>

        
        <li>
          <a class="external-link nav-link" id="css-toggle-btn" aria-label="github">
            <span class="fas fa fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        
        
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="bulkanalyser-an-accessible-interactive-pipeline-for-analysing-and-sharing-bulk-sequencing-results">
<em>bulkAnalyseR</em>: An accessible, interactive pipeline for analysing and sharing bulk sequencing results<a class="anchor" aria-label="anchor" href="#bulkanalyser-an-accessible-interactive-pipeline-for-analysing-and-sharing-bulk-sequencing-results"></a>
</h1></div>

<div style="text-align: justify">
<p>Bulk sequencing experiments (e.g. mRNAseq, sRNAseq etc) are essential for exploring a wide range of biological questions. To bring the data analysis closer to its interpretation and facilitate both interactive, exploratory tasks and the sharing of (easily accessible) information, we present <em>bulkAnalyseR</em>, an R package that offers a seamless, customisable solution for most bulk sequencing datasets. By integrating state-of-the-art approaches without relying on extensive computational support, and replacing static images with interactive panels, our aim is to further support and strengthen the reusability of data. <em>bulkAnalyseR</em> enables standard analyses of bulk data, using an expression matrix as starting point. It presents the outputs of various steps in an interactive web-based interface, making it easy to generate, explore and verify hypotheses. Moreover, the app can be easily shared and published, incentivising research reproducibility and allowing others to explore the same processed data.</p>
<p>Preprint: <a href="https://www.biorxiv.org/content/10.1101/2021.12.23.473982v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/2021.12.23.473982v1</a></p>
<p>Live app example: <a href="http://bioinf.stemcells.cam.ac.uk:3838/bulkAnalyseR/Yang2019/" class="external-link uri">http://bioinf.stemcells.cam.ac.uk:3838/bulkAnalyseR/Yang2019/</a></p>
<p>If you are using components of this package in published research please cite the following papers along with the <em>bulkAnalyseR</em> manuscript:</p>
<ul>
<li>
<strong>Noise removal:</strong> Moutsopoulos, I. et al. (2021). noisyR: enhancing biological signal in sequencing datasets by characterizing random technical noise. Nucleic Acids Research, 49(14), e83–e83.</li>
<li>
<strong>Differential expression:</strong> Robinson, M. D. et al. (2009). edgeR: a Bioconductor package for differential expression analysis of digital gene expression data. Bioinformatics, 26(1), 139–140. and Love M.I., et al. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol. 2014; 15:550.</li>
<li>
<strong>Enrichment analysis:</strong> Raudvere, U. et al. (2019). g:Profiler: a web server for functional enrichment analysis and conversions of gene lists (2019 update). Nucleic Acids Research, 47(W1), W191–W198.</li>
<li>
<strong>Gene regulatory network inference:</strong> Huynh-Thu, V. A. et al. (2010). Inferring regulatory networks from expression data using tree-based methods. PloS one, 5(9), e12776.</li>
</ul>
<p><img src="https://github.com/Core-Bioinformatics/bulkAnalyseR/blob/main/vignettes/figures/workflow.png?raw=true" alt="drawing" width="500"></p>
<p><em>Workflow diagram of the <strong>bulkAnalyseR</strong> pipeline. The input comprises a processed (i.e. normalised, noise-corrected) expression matrix. Using <strong>bulkAnalyseR</strong>, all standard steps related to differential expression analyses are handled seamlessley. The pairwaise comparison of differential expression outputs is also possible (using cross plots and upset plots). Finally, localised Gene Regulatory Networks can be created.</em></p>
<div class="section level2">
<h2 id="preprocessing-step">Preprocessing step<a class="anchor" aria-label="anchor" href="#preprocessing-step"></a>
</h2>
<div class="section level3">
<h3 id="defining-the-input-expression-matrix-and-corresponding-metadata">Defining the input expression matrix and corresponding metadata<a class="anchor" aria-label="anchor" href="#defining-the-input-expression-matrix-and-corresponding-metadata"></a>
</h3>
<p>To create a shiny app using <em>bulkAnalyseR</em>, you need a processed (e.g. normalised, noise corrected) expression matrix and a corresponding metadata table loaded in your workspace.</p>
<p>The expression matrix is expected to have genes on the rows (with the Ensembl ID as the row name) and samples on the columns.</p>
<p>The first column of the metadata table must match the column names of the expression matrix. The other columns of the metadata table contain other information about the samples e.g. time points, treatment groups, demographic information etc; the differential expression analysis may take into account a subset of these columns.</p>
<p>We illustrate this on a case study from a 2019 paper by Yang et al (<a href="https://www.sciencedirect.com/science/article/pii/S2405471219301152" class="external-link uri">https://www.sciencedirect.com/science/article/pii/S2405471219301152</a>). Using three time points, each with two replicates, for some bulk mRNAseq samples, we obtain the following matrices:</p>
<table class="table">
<thead><tr class="header">
<th></th>
<th>SRR7624365</th>
<th>SRR7624366</th>
<th>SRR7624371</th>
<th>SRR7624372</th>
<th>SRR7624375</th>
<th>SRR7624376</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>ENSMUSG00000102693</td>
<td>2</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>ENSMUSG00000064842</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>ENSMUSG00000051951</td>
<td>6</td>
<td>4</td>
<td>2</td>
<td>0</td>
<td>47</td>
<td>37</td>
</tr>
<tr class="even">
<td></td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
<p>Below is an example of a metadata table:</p>
<table class="table">
<thead><tr class="header">
<th>srr</th>
<th>timepoint</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>SRR7624365</td>
<td>0h</td>
</tr>
<tr class="even">
<td>SRR7624366</td>
<td>0h</td>
</tr>
<tr class="odd">
<td>SRR7624371</td>
<td>12h</td>
</tr>
<tr class="even">
<td>SRR7624372</td>
<td>12h</td>
</tr>
<tr class="odd">
<td>SRR7624375</td>
<td>36h</td>
</tr>
<tr class="even">
<td>SRR7624376</td>
<td>36h</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="denoising-and-normalisation">Denoising and normalisation<a class="anchor" aria-label="anchor" href="#denoising-and-normalisation"></a>
</h3>
<p>Before using the expression matrix to create a shiny app, some preprocessing should be performed. <em>bulkAnalyseR</em> contains the function <strong>preprocessExpressionMatrix</strong> which takes the raw expression matrix as input, then denoises the data using <a href="https://github.com/Core-Bioinformatics/noisyR" class="external-link"><em>noisyR</em></a> and normalises the expression levels using either quantile (by default) or RPM normalisation (specified using <em>normalisation.method</em> parameter).</p>
<p>It is not recommended to use data which has not been denoised and normalised as input to <em>generateShinyApp</em>; noisy data is prone to spurious, un-reproducible patterns; analyses performed on unnormalised data are unlikely to be robust. You can also perform your own preprocessing outside <em>preprocessExpressionMatrix</em> function.</p>
<p>Main function: <em>preprocessExpressionMatrix()</em></p>
<p>Supporting function: <em>noisyr_counts_with_plot()</em></p>
</div>
</div>
<div class="section level2">
<h2 id="generating-shiny-app">Generating shiny app<a class="anchor" aria-label="anchor" href="#generating-shiny-app"></a>
</h2>
<p>The central function in <em>bulkAnalyseR</em> is <strong>generateShinyApp</strong>. This function creates an app.R file and all required objects to run the app in .rda format in the target directory. The key inputs to <strong>generateShinyApp</strong> are <em>expression.matrix</em> (after being processed using <em>preprocessExpressionMatrix</em>) and <em>meta</em>. You can also specify the title, a folder name where the app will be saved, a shiny theme, as well as specifying the organism on which your data was generated.</p>
<p>Calling <em>generateShinyApp</em> with these parameters will create a folder with your chosen name in which there will be 2 files <em>expression_matrix.rda</em> and <em>metadata.rda</em> and <em>app.R</em> which defines the app. To see the app, you can call <em>shiny::runApp()</em> with the name of the folder as parameter. The app generated is standalone and can be shared with collaborators or published online through a platform like <a href="https://www.shinyapps.io/" class="external-link">shinyapps.io</a>. This provides an easy way for anyone to explore the data and verify the conclusions, increasing access and promoting reproducibility of the bioinformatics analysis.</p>
<p>By default, the app will have 9 panels: Sample select, Quality checks, Differential expression, Volcano and MA plots, DE summary, Enrichment, Expression patterns, Cross plots, GRN inference. You can choose to remove one or more panels using the <em>default.panels</em> parameter.</p>
<p>By default, the app will look like this:</p>
<p><img src="https://github.com/Core-Bioinformatics/bulkAnalyseR/blob/main/vignettes/figures/ScreenShot.png?raw=true" alt="drawing" width="800"></p>
<p><em>Screenshot from Yang case study processed with the bulkAnalyseR app</em></p>
<p>See <a href="https://raw.githack.com/Core-Bioinformatics/bulkAnalyseR/main/vignettes/bulkAnalyseR.html" class="external-link">vignette</a> for more details on the individual panels.</p>
<p>You can also add custom extra panels and data using the <em>panels.extra</em> and <em>data.extra</em> parameters.</p>
<p>Main function: <em>generateShinyApp()</em></p>
<p>Panel functions (with UI and server components): <em>sampleSelectPanel</em>, <em>QCpanel</em>, <em>DEpanel</em>, <em>DEplotPanel</em>, <em>DEsummaryPanel</em>, <em>enrichmentPanel</em>, <em>patternPanel</em>, <em>crossPanel</em>, <em>GRNpanel</em></p>
<p>Supporting functions: <em>DEanalysis_deseq2</em>, <em>DEanalysis_edger</em>, <em>calculate_condition_mean_sd_per_gene</em>, <em>determine_uds</em>, <em>expression_heatmap</em>, <em>jaccard_heatmap</em>, <em>ma_enhance</em>, <em>ma_plot</em>, <em>make_heatmap_matrix</em>, <em>make_pattern_matrix</em>, <em>plot_line_pattern</em>, <em>plot_pca</em>, <em>volcano_enhance</em>, <em>volcano_plot</em></p>
</div>
<div class="section level2">
<h2 id="quick-start-guide">Quick Start Guide<a class="anchor" aria-label="anchor" href="#quick-start-guide"></a>
</h2>
<p>A shiny app for the <a href="https://www.sciencedirect.com/science/article/pii/S2405471219301152" class="external-link">Yang et al 2019 data</a> can be generated using the following code:</p>
<pre class="{r}"><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Core-Bioinformatics/bulkAnalyseR" class="external-link">bulkAnalyseR</a></span><span class="op">)</span> 

<span class="co">#load expression matrix</span>
<span class="va">expression.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expression_matrix.csv"</span>, package <span class="op">=</span> <span class="st">"bulkAnalyseR"</span><span class="op">)</span>, 
  row.names <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span><span class="op">)</span>

<span class="co">#create metadata table</span>
<span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>srr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">expression.matrix</span><span class="op">)</span>, 
                   timepoint <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"0h"</span>, <span class="st">"12h"</span>, <span class="st">"36h"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span>

<span class="co">#run preprocessing</span>
<span class="va">expression.matrix.preproc</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/preprocessExpressionMatrix.html">preprocessExpressionMatrix</a></span><span class="op">(</span><span class="va">expression.matrix</span><span class="op">)</span>

<span class="co">#create shiny app</span>
<span class="va">shiny.dir</span> <span class="op">&lt;-</span> <span class="st">"shiny_Yang2019"</span>
<span class="fu"><a href="reference/generateShinyApp.html">generateShinyApp</a></span><span class="op">(</span>
  expression.matrix <span class="op">=</span> <span class="va">expression.matrix.preproc</span>,
  metadata <span class="op">=</span> <span class="va">metadata</span>,
  shiny.dir <span class="op">=</span> <span class="va">shiny.dir</span>,
  app.title <span class="op">=</span> <span class="st">"Shiny app for three timepoints from the Yang 2019 data"</span>,
  organism <span class="op">=</span> <span class="st">"mmusculus"</span>,
  org.db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span>
<span class="op">)</span>

<span class="co">#run shiny app</span>
<span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">shiny.dir</span><span class="op">)</span></code></pre>
</div>
<div class="section level2">
<h2 id="installation-guide">Installation guide<a class="anchor" aria-label="anchor" href="#installation-guide"></a>
</h2>
<p><em>bulkAnalyseR</em> can be installed from CRAN using <em>install.packages(“bulkAnalyseR”)</em>. Please make sure all bioconductor dependencies are also installed.</p>
<p>To install the latest stable development version from GitHub, first install the CRAN dependencies as well as <em>devtools</em> then use <em>devtools::install_github(“Core-Bioinformatics/bulkAnalyseR”)</em>.</p>
<div class="section level3">
<h3 id="required-cran-packages-use-installpackages">Required CRAN packages (use <em>install.packages()</em>)<a class="anchor" aria-label="anchor" href="#required-cran-packages-use-installpackages"></a>
</h3>
<ul>
<li>utils</li>
<li>stats</li>
<li>grDevices</li>
<li>tibble</li>
<li>dplyr</li>
<li>magrittr</li>
<li>shiny</li>
<li>shinythemes</li>
<li>shinyWidgets</li>
<li>shinyjqui</li>
<li>shinyjs</li>
<li>ggplot2</li>
<li>ggrepel</li>
<li>ggnewscale</li>
<li>ggforce</li>
<li>ggrastr</li>
<li>RColorBrewer</li>
<li>glue</li>
<li>rlang</li>
<li>noisyr</li>
<li>matrixStats</li>
<li>visNetwork</li>
<li>gprofiler2</li>
<li>circlize</li>
<li>shinyLP</li>
</ul>
</div>
<div class="section level3">
<h3 id="required-bioconductor-packages-use-biocmanagerinstall">Required Bioconductor packages (use <em>BiocManager::install()</em>)<a class="anchor" aria-label="anchor" href="#required-bioconductor-packages-use-biocmanagerinstall"></a>
</h3>
<ul>
<li>preprocessCore</li>
<li>edgeR</li>
<li>DESeq2</li>
<li>AnnotationDBI</li>
<li>GENIE3</li>
<li>ComplexHeatmap</li>
</ul>
</div>
<div class="section level3">
<h3 id="bioconductor-annotation-packages-the-one-for-your-model-organism-is-required-human-and-mouse-ones-are-listed-here">Bioconductor annotation packages (the one for your model organism is required, human and mouse ones are listed here)<a class="anchor" aria-label="anchor" href="#bioconductor-annotation-packages-the-one-for-your-model-organism-is-required-human-and-mouse-ones-are-listed-here"></a>
</h3>
<ul>
<li>org.Hs.eg.db</li>
<li>org.Mm.eg.db</li>
</ul>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=bulkAnalyseR" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/Core-Bioinformatics/bulkAnalyseR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/Core-Bioinformatics/bulkAnalyseR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://www.r-project.org/Licenses/GPL-2" class="external-link">GPL-2</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing bulkAnalyseR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Ilias Moutsopoulos <br><small class="roles"> Author, maintainer </small>  </li>
<li>Eleanor Williams <br><small class="roles"> Author, contributor </small>  </li>
<li>Irina Mohorianu <br><small class="roles"> Author, contributor </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="http://cranlogs.r-pkg.org/badges/bulkAnalyseR" alt="rstudio mirror downloads"></a></li>
<li><a href="https://github.com/r-hub/cranlogs.app" class="external-link"><img src="http://cranlogs.r-pkg.org/badges/grand-total/bulkAnalyseR" alt="rstudio mirror downloads"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="copyright">
  <p></p>
<p>Developed by Ilias Moutsopoulos, Eleanor Williams, Irina Mohorianu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
