<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="bulkAnalyseR">
<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/5/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>bulkAnalyseR: An easy way to analyse and share bulk mRNA-Seq data • bulkAnalyseR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="bulkAnalyseR: An easy way to analyse and share bulk mRNA-Seq data">
<meta property="og:description" content="bulkAnalyseR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/5/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/5/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">
<!-- preferably CSS --><link rel="stylesheet" href="../preferably.css">
<link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">bulkAnalyseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/bulkAnalyseR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Core-Bioinformatics/bulkAnalyseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>

        
        
        
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>bulkAnalyseR: An easy way to analyse and share bulk mRNA-Seq data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Core-Bioinformatics/bulkAnalyseR/blob/HEAD/vignettes/bulkAnalyseR.Rmd" class="external-link"><code>vignettes/bulkAnalyseR.Rmd</code></a></small>
      <div class="d-none name"><code>bulkAnalyseR.Rmd</code></div>
    </div>

    
    
<div style="text-align: justify">



<p>Bulk mRNA-seq experiments are essential for exploring a wide range of biological questions. To bring the data analysis closer to its interpretation and facilitate both interactive, exploratory tasks and the sharing of (easily accessible) information, we present <em>bulkAnalyseR</em> an R package that offers a seamless, customisable solution for most bulk RNAseq datasets. By integrating state-of-the-art approaches without relying on extensive computational support, and replacing static images with interactive panels, our aim is to further support and strengthen the reusability of data. bulkAnalyseR enables standard analyses of bulk data, using an expression matrix as starting point. It presents the outputs of various steps in an interactive web-based interface, making it easy to generate, explore and verify hypotheses. Moreover, the app can be easily shared and published, incentivising research reproducibility and allowing others to explore the same processed data and enhance the biological conclusions.</p>
<p><img src="figures/workflow.png" width="80%"></p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the package, first install all CRAN dependencies:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">packages.cran</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ggplot2"</span>,
                   <span class="st">"shiny"</span>,
                   <span class="st">"shinythemes"</span>,
                   <span class="st">"gprofiler2"</span>,
                   <span class="st">"stats"</span>,
                   <span class="st">"ggrepel"</span>,
                   <span class="st">"utils"</span>,
                   <span class="st">"RColorBrewer"</span>,
                   <span class="st">"circlize"</span>,
                   <span class="st">"shinyWidgets"</span>,
                   <span class="st">"shinyjqui"</span>,
                   <span class="st">"dplyr"</span>,
                   <span class="st">"magrittr"</span>,
                   <span class="st">"ggforce"</span>,
                   <span class="st">"rlang"</span>,
                   <span class="st">"glue"</span>,
                   <span class="st">"matrixStats"</span>,
                   <span class="st">"noisyr"</span>,
                   <span class="st">"tibble"</span>,
                   <span class="st">"ggnewscale"</span>,
                   <span class="st">"ggrastr"</span>,
                   <span class="st">"visNetwork"</span><span class="op">)</span>
<span class="va">new.packages.cran</span> <span class="op">&lt;-</span> <span class="va">packages.cran</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">packages.cran</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>, <span class="st">"Package"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span>
<span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">new.packages.cran</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">new.packages.cran</span><span class="op">)</span></code></pre></div>
<p>Then install bioconductor dependencies:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">packages.bioc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"edgeR"</span>, 
                   <span class="st">"DESeq2"</span>,
                   <span class="st">"preprocessCore"</span>,
                   <span class="st">"AnnotationDbi"</span>,
                   <span class="st">"GENIE3"</span>,
                   <span class="st">"ComplexHeatmap"</span><span class="op">)</span>

<span class="va">new.packages.bioc</span> <span class="op">&lt;-</span> <span class="va">packages.bioc</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">packages.bioc</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>,<span class="st">"Package"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span>
<span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">new.packages.bioc</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">new.packages.bioc</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p>Finally, you can install the latest stable version of <em>bulkAnalyseR</em> from GitHub:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"Core-Bioinformatics/bulkAnalyseR"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h2>
<p>First, load <em>bulkAnalyseR</em>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Core-Bioinformatics/bulkAnalyseR" class="external-link">bulkAnalyseR</a></span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="loading-an-expression-matrix">Loading an expression matrix<a class="anchor" aria-label="anchor" href="#loading-an-expression-matrix"></a>
</h3>
<p>For this vignette we are using a subset of the count matrix for an experiment included in <a href="https://www.sciencedirect.com/science/article/pii/S2405471219301152" class="external-link">a 2019 paper by Yang et al</a>. Rows represent genes/features and columns represent samples:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts.in</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expression_matrix.csv"</span>, package <span class="op">=</span> <span class="st">"bulkAnalyseR"</span><span class="op">)</span>
<span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">counts.in</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span>
<span class="co">##&gt;                    SRR7624365 SRR7624366 SRR7624371 SRR7624372 SRR7624375</span>
<span class="co">##&gt; ENSMUSG00000102693          2          0          0          0          0</span>
<span class="co">##&gt; ENSMUSG00000051951          6          4          2          0         47</span>
<span class="co">##&gt; ENSMUSG00000102851          0          0          0          0          0</span>
<span class="co">##&gt; ENSMUSG00000103377          0          0          2          0          8</span>
<span class="co">##&gt; ENSMUSG00000104017          0          0          0          2          0</span>
<span class="co">##&gt; ENSMUSG00000103025          0          0          0          0          0</span>
<span class="co">##&gt;                    SRR7624376</span>
<span class="co">##&gt; ENSMUSG00000102693          0</span>
<span class="co">##&gt; ENSMUSG00000051951         37</span>
<span class="co">##&gt; ENSMUSG00000102851          1</span>
<span class="co">##&gt; ENSMUSG00000103377          6</span>
<span class="co">##&gt; ENSMUSG00000104017          2</span>
<span class="co">##&gt; ENSMUSG00000103025          3</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="defining-metadata">Defining metadata<a class="anchor" aria-label="anchor" href="#defining-metadata"></a>
</h3>
<p>The subset of the dataset we are using has samples from 3 timepoints: 0h, 12h and 36h, each with 2 biological replicates. We define a metadata table detailing which sample correspond to which timepoint:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  srr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span>, 
  timepoint <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"0h"</span>, <span class="st">"12h"</span>, <span class="st">"36h"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>This metadata table should be a data frame containing at minimum two columns: the first column must contain the column names of the expression.matrix, while the last column is assumed to contain the experimental conditions that will be tested for differential expression.</p>
</div>
<div class="section level3">
<h3 id="denoising-and-normalisation">Denoising and normalisation<a class="anchor" aria-label="anchor" href="#denoising-and-normalisation"></a>
</h3>
<p>Before using the expression matrix to create our shiny app, some preprocessing should be performed. <em>bulkAnalyseR</em> contains the function <strong>preprocessExpressionMatrix</strong> which takes the expression matrix as input then denoises the data using <a href="https://github.com/Core-Bioinformatics/noisyR" class="external-link"><em>noisyR</em></a> and normalises using either quantile (by default) or RPM normalisation (specified using <em>normalisation.method</em> parameter). By specifying <em>output.plot = TRUE</em>, you can also print the expression-similarity line plots from <em>noisyR</em> to console and you can specify further parameters from the noisyR <a href="https://core-bioinformatics.github.io/noisyR/reference/noisyr_counts.html" class="external-link"><em>noisyr_counts</em></a>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exp.proc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocessExpressionMatrix.html">preprocessExpressionMatrix</a></span><span class="op">(</span><span class="va">exp</span>, output.plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">##&gt; &gt;&gt;&gt; noisyR counts approach pipeline &lt;&lt;&lt;</span>
<span class="co">##&gt; The input matrix has 38705 rows and 6 cols</span>
<span class="co">##&gt;     number of genes: 38705</span>
<span class="co">##&gt;     number of samples: 6</span>
<span class="co">##&gt; Calculating the number of elements per window</span>
<span class="co">##&gt;     the number of elements per window is 3870</span>
<span class="co">##&gt;     the step size is 193</span>
<span class="co">##&gt;     the selected similarity metric is correlation_pearson</span>
<span class="co">##&gt;   Working with sample 1</span>
<span class="co">##&gt;   Working with sample 2</span>
<span class="co">##&gt;   Working with sample 3</span>
<span class="co">##&gt;   Working with sample 4</span>
<span class="co">##&gt;   Working with sample 5</span>
<span class="co">##&gt;   Working with sample 6</span>
<span class="co">##&gt; Calculating noise thresholds for 6 samples...</span>
<span class="co">##&gt;     similarity.threshold = 0.25</span>
<span class="co">##&gt;     method.chosen = Boxplot-IQR</span>
<span class="co">##&gt; Denoising expression matrix...</span>
<span class="co">##&gt;     removing noisy genes</span>
<span class="co">##&gt;     adjusting matrix</span>
<span class="co">##&gt; &gt;&gt;&gt; Done! &lt;&lt;&lt;</span>
<span class="co">##&gt; Performing quantile normalisation...</span>
<span class="co">##&gt; Done!</span></code></pre></div>
<p><img src="bulkAnalyseR_files/figure-html/preprocess-1.png" width="672"></p>
<p>It is not recommended to use data which has not been denoised and normalised as input to <em>generateShinyApp</em>. You can also perform your own preprocessing outside <em>preprocessExpressionMatrix</em>.</p>
</div>
</div>
<div class="section level2">
<h2 id="creating-a-shiny-app">Creating a shiny app<a class="anchor" aria-label="anchor" href="#creating-a-shiny-app"></a>
</h2>
<p>The central function in <em>bulkAnalyseR</em> is <strong>generateShinyApp</strong>. This function creates an app.R file and all required objects to run the app in .rda format in the target directory. The key inputs to <strong>generateShinyApp</strong> are <em>expression.matrix</em> (after being processed using <em>preprocessExpressionMatrix</em>) and <em>metadata</em>. You can also specify the title of the app (which will appear in the navigation bar at the top of the app) with <em>app.title</em>, the directory where the app should be saved with <em>shiny.dir</em> and the shiny theme you wish to use (‘flatly’ is the default, you can find the other options <a href="https://rstudio.github.io/shinythemes/" class="external-link">here</a>). It is also recommended that you specify the organism on which your data was generated, firstly using the <em>organism</em> parameter using the <em>gprofiler2</em> naming convention e.g. ‘hsapiens’,‘mmusculus’ (see <a href="https://biit.cs.ut.ee/gprofiler/page/organism-list" class="external-link">here</a> for the full list of organisms and IDs), and secondly specifying the database for annotations to convert ENSEMBL IDs to gene names e.g. org.Hs.eg.db - the full list of bioconductor packaged databases can be seen using this command:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/available.html" class="external-link">available</a></span><span class="op">(</span><span class="st">"^org\\."</span><span class="op">)</span></code></pre></div>
<p>The dataset in this example was generated on <em>M. musculus</em> so we would generate the app using this function call (note that the org.Mm.eg.db needs to be installed):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/generateShinyApp.html">generateShinyApp</a></span><span class="op">(</span>
  expression.matrix <span class="op">=</span> <span class="va">exp.proc</span>,
  metadata <span class="op">=</span> <span class="va">meta</span>,
  shiny.dir <span class="op">=</span> <span class="st">"shiny_Yang2019"</span>,
  app.title <span class="op">=</span> <span class="st">"Shiny app for visualisation of three timepoints from the Yang 2019 data"</span>,
  organism <span class="op">=</span> <span class="st">"mmusculus"</span>,
  org.db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span>
<span class="op">)</span></code></pre></div>
<p>If no organism is specified for g:profiler then the enrichment tab will be automatically excluded. If no model organism database is specified then the row names of the expression matrix will be used throughout. It is recommended to use ENSEMBL ids as the row names and supply the model organism through the org.db parameter if your organism is among the ones provided, to ensure compatibility.</p>
<p>This will create a folder called <em>shiny_Yang2019</em> in which there will be 2 data files <em>expression_matrix.rda</em> and <em>metadata.rda</em> and <em>app.R</em> which defines the app. To see the app, you can call <em>shiny::runApp(‘shiny_Yang2019’)</em> and the app will start. The app generated is standalone and can be shared with collaborators or published online through a platform like . This provides an easy way for anyone to explore the data and verify the conclusions, increasing access and promoting reproducibility of the bioinformatics analysis.</p>
<p>By default, the app will have 9 panels: Sample select, Quality checks, Differential expression, Volcano and MA plots, DE summary, Enrichment, Expression patterns, Cross plots, GRN inference. You can choose to remove one or more panels using the <em>panels.default</em> parameter.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/generateShinyApp.html">generateShinyApp</a></span><span class="op">(</span>
  expression.matrix <span class="op">=</span> <span class="va">exp.proc</span>,
  metadata <span class="op">=</span> <span class="va">meta</span>,
  shiny.dir <span class="op">=</span> <span class="st">"shiny_Yang2019_onlyQC_DE"</span>,
  app.title <span class="op">=</span> <span class="st">"Shiny app for visualisation of three timepoints from the Yang 2019 data"</span>,
  organism <span class="op">=</span> <span class="st">"mmusculus"</span>,
  org.db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span>,
  panels.default <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'QC'</span>,<span class="st">'DE'</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>See the following sections for more details about the default panels:</p>
<div class="section level3">
<h3 id="sample-select">Sample Select<a class="anchor" aria-label="anchor" href="#sample-select"></a>
</h3>
<p>This tab allows you to select a subset of samples to use for the analysis. In particular this allows you to exclude low quality samples (which you could discover using QC tab) or focus on particular parts of the data. Once you have selected the samples you want, make sure to press the ‘Use the selected sample!’ button to confirm your choice. If you do not use this button or select samples then all samples will be used.</p>
<p><img src="figures/ScreenShot.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="quality-check-qc">Quality check (QC)<a class="anchor" aria-label="anchor" href="#quality-check-qc"></a>
</h3>
<p>The Quality Check (QC) tab includes a Jaccard Similarity Index (JSI) heatmap, and a PCA dimensionality reduction, with groups based on the metadata information. This enables a high-level overview of the similarity across samples, usually reflecting the experimental design quite closely and serving as a sanity check. Within this panel, there is also the option to show the MA plot between any 2 columns - this can be used to further investigate any sample similarity or differences found through the JSI and PCA plots.</p>
<div class="section level4">
<h4 id="jaccard-similarity-index">Jaccard Similarity Index<a class="anchor" aria-label="anchor" href="#jaccard-similarity-index"></a>
</h4>
<p>Within the JSI plot, you can change the order of samples using an ordering of metadata columns as well as specifying the number of top abundance genes that should be used to calculate the JSI. Finally, you can choose whether JSI values should be shown on the heatmap.</p>
<p><img src="figures/JSI.png" width="80%"></p>
</div>
<div class="section level4">
<h4 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h4>
<p>The PCA plot can be created for a chosen number of top abundance genes and coloured by the metadata attribute selected. The sample labels and ellipses around groups can also be included or removed.</p>
<p><img src="figures/PCA.png" width="80%"></p>
</div>
<div class="section level4">
<h4 id="ma-plots">MA plots<a class="anchor" aria-label="anchor" href="#ma-plots"></a>
</h4>
<p>The third part of the QC tab allows you to create MA plots comparing any two samples. You could, for example, use these plots to compare replicates - these would be expected to show a close funneling shape.</p>
<p><img src="figures/MA_QC.png" width="80%"></p>
</div>
</div>
<div class="section level3">
<h3 id="differential-expression-de">Differential expression (DE)<a class="anchor" aria-label="anchor" href="#differential-expression-de"></a>
</h3>
<p>The differential expression (DE) tab can be used to compare any of the conditions from the columns of the metadata table using the edgeR or DESeq2 packages with customisable parameters (log2 fold change and adjusted p-value threshold). The output is a list of differentially expressed genes along with log2 fold changes and adjusted p-values. Within the app, the differentially expressed genes can be explored using an interactive table (sorted by absolute log2 fold change) and this can also be saved as a comma-separated file.</p>
<p><img src="figures/DE.png" width="80%"></p>
<p>In this tab, you can also select genes of interest by clicking on the table, which can be used in the DE summary and Volcano/MA tabs. There is also a button to choose the top 50 DE genes (by absolute log2 fold change) or reset the selection.</p>
</div>
<div class="section level3">
<h3 id="volcanoma-plots">Volcano/MA plots<a class="anchor" aria-label="anchor" href="#volcanoma-plots"></a>
</h3>
<p>Assuming differential expression has been performed, the Volcano/MA panel then visualises the DE genes, allowing interactive exploration of the data by searching for specific genes of interest or using the selected top DE genes and clicking on genes on the plot itself to gain more information and generate hypotheses.</p>
<div class="section level4">
<h4 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h4>
<p>Volcano plots show log2 fold change against log10 BH-adjusted p-value. The DE points are shown in blue with their colour proportional to the log2 expression of the gene. Green and blue guide lines show the DE thresholds and twice the DE thresholds respectively.</p>
<p><img src="figures/VolcanoPlot.png" width="80%"></p>
</div>
<div class="section level4">
<h4 id="ma-plots-1">MA plots<a class="anchor" aria-label="anchor" href="#ma-plots-1"></a>
</h4>
<p>The DE MA plots show log2 expression levels against log2 fold change for each gene. The DE points are shown in blue with their colour proportional to the log2 expression of the gene. Green and blue guide lines show the DE thresholds and twice the DE thresholds respectively.</p>
<p><img src="figures/MAPlot.png" width="80%"></p>
</div>
</div>
<div class="section level3">
<h3 id="de-summary">DE summary<a class="anchor" aria-label="anchor" href="#de-summary"></a>
</h3>
<p>The DE summary panel allows further visualisation using the genes called DE or selected by the user in the DE panel.</p>
<div class="section level4">
<h4 id="post-differential-expression-pca">Post-differential expression PCA<a class="anchor" aria-label="anchor" href="#post-differential-expression-pca"></a>
</h4>
<p>Firstly, we can create a PCA using only the DE genes or using the selected genes from the DE panel. You can also create the PCA on only the user-selected DE genes by deselecting the ‘Use all DE?’ option. As in the QC PCA plot, you can colour points by your chosen metadata columns and add/remove sample labels and group ellipses.</p>
<p><img src="figures/PCA_DE.png" width="80%"></p>
</div>
<div class="section level4">
<h4 id="expression-heatmap">Expression heatmap<a class="anchor" aria-label="anchor" href="#expression-heatmap"></a>
</h4>
<p>Secondly, we can create heatmaps for expression levels (raw expression level, log2 expression level or Z-score) of selected genes. By default the top 50 DE genes are used, but if users select genes in the DE panel then these are used instead. Also, further genes can be added by name in the Additional genes box. As in the JSI, you can change the order of samples using an ordering of metadata columns.</p>
<p><img src="figures/ExpressionHeatmap.png" width="80%"></p>
</div>
</div>
<div class="section level3">
<h3 id="enrichment">Enrichment<a class="anchor" aria-label="anchor" href="#enrichment"></a>
</h3>
<p>The enrichment tab uses the differentially expressed genes to run an enrichment analysis using g:profiler with a user selected output of the standard GO terms, the KEGG and reactome pathway databases, and miRNA and transcription factor lists. The plot shows the log10 BH-corrected p-value for each term, separated by category. The background set used is all the genes in the expression matrix - if noisyR preprocessing is used then this will be all the genes above the s/n threshold.</p>
<p>The output is both the results table from g:profiler which can be downloaded as well as a jitter plot which can be clicked to identify the most significant terms.</p>
<p><img src="figures/Enrichment.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="expression-patterns">Expression patterns<a class="anchor" aria-label="anchor" href="#expression-patterns"></a>
</h3>
<p>The expression pattern tab allows the creation of expression patterns to identify potential genes of interest across a variety of conditions. The most common application of this is a time series, but it could be suitable for another logical progression between conditions. To define the series, the user must select a column of the metadata and drag states into the “Series of states to use” area.</p>
<p>The pattern identification is done by calculating a confidence interval for each gene in each condition, using all samples in that condition and the number of standard deviations away from the mean provided. The pattern between two consecutive conditions is defined as straight (S) if the intervals overlap and up (U) or down (D) if they don’t. The full expression pattern is the concatenation of individual patterns (for example, “UUS” for 4 conditions).</p>
<p>The grouped expression matrix can then be downloaded, showing which pattern each gene was assigned to. Plots are also created for the genes in the selected pattern (“Pattern to plot”).</p>
<p><img src="figures/ExpressionPatterns.png" width="30%"></p>
<div class="section level4">
<h4 id="pattern-line-plot">Pattern line plot<a class="anchor" aria-label="anchor" href="#pattern-line-plot"></a>
</h4>
<p>A line plot is shown with the mean expression of the genes assigned to the chosen pattern in each condition. The expression values are mean-scaled by default. A legend is shown if less than 10 genes are present.</p>
<p><img src="figures/ExpressionPatternsLinePlot.png" width="80%"></p>
</div>
<div class="section level4">
<h4 id="pattern-expression-heatmap">Pattern expression heatmap<a class="anchor" aria-label="anchor" href="#pattern-expression-heatmap"></a>
</h4>
<p>A heatmap is shown with the mean expression of the genes assigned to the chosen pattern in each condition. The expression values are z-score transformed by default. Gene names are shown if less than 50 genes are present.</p>
<p><img src="figures/ExpressionPatternsHeatmap.png" width="80%"></p>
</div>
</div>
<div class="section level3">
<h3 id="cross-plots">Cross plots<a class="anchor" aria-label="anchor" href="#cross-plots"></a>
</h3>
<p>The cross plot tab allows you to compare two differential expression analyses against each other, for example two comparisons of interest or the same comparison using edgeR and DESeq2. The plot shows the log2 fold change of the two differential expression calls on each axis. Genes which are DE in both comparisons are coloured purple, in comparison 1 but not comparison 2 in blue and in comparison 2 but not comparison 1 in red. You can label selected genes and click on genes on the plot itself to gain more information and generate hypotheses.</p>
<p><img src="figures/CrossPlot.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="gene-regulatory-networks-grn">Gene regulatory networks (GRN)<a class="anchor" aria-label="anchor" href="#gene-regulatory-networks-grn"></a>
</h3>
<p>The GRN tab enables the creation of small gene regulatory networks (GRNs) to facilitate further exploration and hypothesis generation based on genes of interest. Target genes can be selected and a small network with them as targets can be generated by clicking the “Start GRN inference” button. The number of regulators can then be adjusted and the plot downloaded in interactive html format. Note that target genes can also regulate each other if the selected genes are functionally similar.</p>
<p><img src="figures/GRN.png" width="80%"></p>
</div>
<div class="section level3">
<h3 id="adding-extra-panels">Adding extra panels<a class="anchor" aria-label="anchor" href="#adding-extra-panels"></a>
</h3>
<p>Alongside the default 8 panels, you can also define your own panels and add them to the app. As an example, we could add an extra QC panel (in this case it will be exactly the same):</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/generateShinyApp.html">generateShinyApp</a></span><span class="op">(</span>
  expression.matrix <span class="op">=</span> <span class="va">exp.proc</span>,
  metadata <span class="op">=</span> <span class="va">meta</span>,
  shiny.dir <span class="op">=</span> <span class="st">"shiny_Yang2019_ExtraQC"</span>,
  app.title <span class="op">=</span> <span class="st">"Shiny app for visualisation of three timepoints from the Yang 2019 data - extra QC"</span>,
  organism <span class="op">=</span> <span class="st">"mmusculus"</span>,
  org.db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span>,
  panels.extra <span class="op">=</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>
    UIfun <span class="op">=</span> <span class="st">"QCpanelUI"</span>, 
    UIvars <span class="op">=</span> <span class="st">"'QC2', metadata"</span>, 
    serverFun <span class="op">=</span> <span class="st">"QCpanelServer"</span>, 
    serverVars <span class="op">=</span> <span class="st">"'QC2', expression.matrix, metadata"</span>
  <span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p>If you need to add extra data or package imports for the extra panel(s) then you can do this using the <em>data.extra</em> and <em>packages.extra</em> parameters. Make sure you have the extra data loaded when you create the app. For example:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">extra.data1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">36</span><span class="op">)</span>,nrow<span class="op">=</span><span class="fl">6</span><span class="op">)</span>
<span class="va">extra.data2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">60</span><span class="op">)</span>,nrow<span class="op">=</span><span class="fl">10</span><span class="op">)</span>

<span class="fu"><a href="../reference/generateShinyApp.html">generateShinyApp</a></span><span class="op">(</span>
  expression.matrix <span class="op">=</span> <span class="va">exp.proc</span>,
  metadata <span class="op">=</span> <span class="va">meta</span>,
  shiny.dir <span class="op">=</span> <span class="st">"shiny_Yang2019_ExtraData"</span>,
  app.title <span class="op">=</span> <span class="st">"Shiny app for visualisation of three timepoints from the Yang 2019 data - extra QC"</span>,
  organism <span class="op">=</span> <span class="st">"mmusculus"</span>,
  org.db <span class="op">=</span> <span class="st">"org.Mm.eg.db"</span>,
  panels.extra <span class="op">=</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>
    UIfun <span class="op">=</span> <span class="st">"QCpanelUI"</span>, 
    UIvars <span class="op">=</span> <span class="st">"'QC2', metadata"</span>, 
    serverFun <span class="op">=</span> <span class="st">"QCpanelServer"</span>, 
    serverVars <span class="op">=</span> <span class="st">"'QC2', expression.matrix, metadata"</span>
  <span class="op">)</span>,
  data.extra <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"extra.data1"</span>, <span class="st">"extra.data2"</span><span class="op">)</span>,
  packages.extra <span class="op">=</span> <span class="st">"somePackage"</span>,
<span class="op">)</span></code></pre></div>
</div>
</div>
  
   <aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="copyright">
  <p></p>
<p>Developed by Ilias Moutsopoulos, Eleanor Williams, Irina Mohorianu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

    </footer></main>
</div>

  

  

  </div>
</body>
</html>
