<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="bulkAnalyseR">
<!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light">
<script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/5/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>A beginner's guide to creating a bulkAnalyseR app from a GEO dataset â€¢ bulkAnalyseR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A beginner's guide to creating a bulkAnalyseR app from a GEO dataset">
<meta property="og:description" content="bulkAnalyseR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/5/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/5/darkly/bootstrap.css" media="(prefers-color-scheme: dark)">
<!-- preferably CSS --><link rel="stylesheet" href="../preferably.css">
<link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)">
<link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)">
<script src="../darkswitch.js"></script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">bulkAnalyseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/bulkAnalyseR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GEO_App.html">A beginner's guide to creating a bulkAnalyseR app from a GEO dataset</a>
    <a class="dropdown-item" href="../articles/IntegratingMultiomics.html">bulkAnalyseR: Integrating multi-omics or external data</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Core-Bioinformatics/bulkAnalyseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>

        
        <li>
          <a class="external-link nav-link" id="css-toggle-btn" aria-label="github">
            <span class="fas fa fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        
        
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>A beginner's guide to creating a bulkAnalyseR app from a GEO dataset</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Core-Bioinformatics/bulkAnalyseR/blob/HEAD/vignettes/GEO_App.Rmd" class="external-link"><code>vignettes/GEO_App.Rmd</code></a></small>
      <div class="d-none name"><code>GEO_App.Rmd</code></div>
    </div>

    
    
<div style="text-align: justify">



<p>In this short tutorial we showcase a simple pipeline to create a
bulkAnalyseR app using a publicly available dataset from the <a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">Gene Expression Omnibus
(GEO)</a>. No pre-requisites are required, as the installation of
bulkAnalyseR and download of the data are included.</p>
<p>The example app described in this vignette can be found <a href="https://bioinf.stemcells.cam.ac.uk/shiny/bulkAnalyseR/GEO/" class="external-link">here</a>.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>First, install the latest version of bulkAnalyseR, starting with the
CRAN and Bioconductor dependencies:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">packages.cran</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"ggplot2"</span>, <span class="st">"shiny"</span>, <span class="st">"shinythemes"</span>, <span class="st">"gprofiler2"</span>, <span class="st">"stats"</span>, <span class="st">"ggrepel"</span>,</span>
<span>  <span class="st">"utils"</span>, <span class="st">"RColorBrewer"</span>, <span class="st">"circlize"</span>, <span class="st">"shinyWidgets"</span>, <span class="st">"shinyjqui"</span>,</span>
<span>  <span class="st">"dplyr"</span>, <span class="st">"magrittr"</span>, <span class="st">"ggforce"</span>, <span class="st">"rlang"</span>, <span class="st">"glue"</span>, <span class="st">"matrixStats"</span>,</span>
<span>  <span class="st">"noisyr"</span>, <span class="st">"tibble"</span>, <span class="st">"ggnewscale"</span>, <span class="st">"ggrastr"</span>, <span class="st">"visNetwork"</span>, <span class="st">"shinyLP"</span>,</span>
<span>  <span class="st">"grid"</span>, <span class="st">"DT"</span>, <span class="st">"scales"</span>, <span class="st">"shinyjs"</span>, <span class="st">"tidyr"</span>, <span class="st">"UpSetR"</span>, <span class="st">"ggVennDiagram"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">new.packages.cran</span> <span class="op">&lt;-</span> <span class="va">packages.cran</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">packages.cran</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>, <span class="st">"Package"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">new.packages.cran</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="va">new.packages.cran</span><span class="op">)</span></span>
<span></span>
<span><span class="va">packages.bioc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"edgeR"</span>, <span class="st">"DESeq2"</span>, <span class="st">"preprocessCore"</span>, <span class="st">"GENIE3"</span>, <span class="st">"ComplexHeatmap"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">new.packages.bioc</span> <span class="op">&lt;-</span> <span class="va">packages.bioc</span><span class="op">[</span><span class="op">!</span><span class="op">(</span><span class="va">packages.bioc</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span>,<span class="st">"Package"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">new.packages.bioc</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">new.packages.bioc</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"bulkAnalyseR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="download-data-and-create-app">Download data and create app<a class="anchor" aria-label="anchor" href="#download-data-and-create-app"></a>
</h2>
<div class="section level3">
<h3 id="get-the-expression-matrix">Get the expression matrix<a class="anchor" aria-label="anchor" href="#get-the-expression-matrix"></a>
</h3>
<p>We start by downloading and reading in the expression matrix. Rows
represent genes/features and columns represent samples (note you need an
internet connection to run the code below). The matrix is from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE178620" class="external-link">a
2022 study</a> on the Stem Cell transcriptional response to
Microglia-Conditioned Media. We only use a few samples in the study for
illustrative purposes.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">download_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"expression_matrix.csv.gz"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE178620&amp;format=file&amp;file=GSE178620%5Fraw%5Fabundances%2Ecsv%2Egz"</span>, </span>
<span>  <span class="va">download_path</span></span>
<span><span class="op">)</span></span>
<span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">download_path</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">19</span>,<span class="fl">20</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span></span>
<span><span class="co">##&gt;                 control_G322_G322_1 control_G322_G322_2 microglia_067MG_G322_1</span></span>
<span><span class="co">##&gt; ENSG00000223972                   0                   0                      0</span></span>
<span><span class="co">##&gt; ENSG00000227232                  51                  45                     25</span></span>
<span><span class="co">##&gt; ENSG00000278267                   6                   0                      0</span></span>
<span><span class="co">##&gt; ENSG00000243485                   0                   0                      0</span></span>
<span><span class="co">##&gt; ENSG00000284332                   0                   0                      0</span></span>
<span><span class="co">##&gt; ENSG00000237613                   0                   0                      0</span></span>
<span><span class="co">##&gt;                 microglia_067MG_G322_2</span></span>
<span><span class="co">##&gt; ENSG00000223972                      0</span></span>
<span><span class="co">##&gt; ENSG00000227232                     40</span></span>
<span><span class="co">##&gt; ENSG00000278267                      0</span></span>
<span><span class="co">##&gt; ENSG00000243485                      0</span></span>
<span><span class="co">##&gt; ENSG00000284332                      0</span></span>
<span><span class="co">##&gt; ENSG00000237613                      0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="defining-metadata">Defining metadata<a class="anchor" aria-label="anchor" href="#defining-metadata"></a>
</h3>
<p>We use a very simple metadata table with just the main condition in
the experiment. Detailed metadata is available for all GEO datasets and
can be downloaded and used instead.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span>,</span>
<span>  condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exp</span><span class="op">)</span>, USE.NAMES <span class="op">=</span> <span class="cn">FALSE</span>, <span class="kw">function</span><span class="op">(</span><span class="va">nm</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">nm</span>, <span class="st">"_"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">meta</span></span>
<span><span class="co">##&gt;                     name condition</span></span>
<span><span class="co">##&gt; 1    control_G322_G322_1   control</span></span>
<span><span class="co">##&gt; 2    control_G322_G322_2   control</span></span>
<span><span class="co">##&gt; 3 microglia_067MG_G322_1 microglia</span></span>
<span><span class="co">##&gt; 4 microglia_067MG_G322_2 microglia</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="pre-processing">Pre-processing<a class="anchor" aria-label="anchor" href="#pre-processing"></a>
</h3>
<p>We can now denoise and normalise the data using bulkAnalyseR</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exp.proc</span> <span class="op">&lt;-</span> <span class="fu">bulkAnalyseR</span><span class="fu">::</span><span class="fu"><a href="../reference/preprocessExpressionMatrix.html">preprocessExpressionMatrix</a></span><span class="op">(</span><span class="va">exp</span>, output.plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">##&gt; &gt;&gt;&gt; noisyR counts approach pipeline &lt;&lt;&lt;</span></span>
<span><span class="co">##&gt; The input matrix has 60671 rows and 4 cols</span></span>
<span><span class="co">##&gt;     number of genes: 60671</span></span>
<span><span class="co">##&gt;     number of samples: 4</span></span>
<span><span class="co">##&gt; Calculating the number of elements per window</span></span>
<span><span class="co">##&gt;     the number of elements per window is 6067</span></span>
<span><span class="co">##&gt;     the step size is 303</span></span>
<span><span class="co">##&gt;     the selected similarity metric is correlation_pearson</span></span>
<span><span class="co">##&gt;   Working with sample 1</span></span>
<span><span class="co">##&gt;   Working with sample 2</span></span>
<span><span class="co">##&gt;   Working with sample 3</span></span>
<span><span class="co">##&gt;   Working with sample 4</span></span>
<span><span class="co">##&gt; Calculating noise thresholds for 4 samples...</span></span>
<span><span class="co">##&gt;     similarity.threshold = 0.25</span></span>
<span><span class="co">##&gt;     method.chosen = Boxplot-IQR</span></span>
<span><span class="co">##&gt; Denoising expression matrix...</span></span>
<span><span class="co">##&gt;     removing noisy genes</span></span>
<span><span class="co">##&gt;     adjusting matrix</span></span>
<span><span class="co">##&gt; &gt;&gt;&gt; Done! &lt;&lt;&lt;</span></span>
<span><span class="co">##&gt; Performing quantile normalisation...</span></span>
<span><span class="co">##&gt; Done!</span></span></code></pre></div>
<p><img src="GEO_App_files/figure-html/preprocess-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="creating-the-shiny-app">Creating the shiny app<a class="anchor" aria-label="anchor" href="#creating-the-shiny-app"></a>
</h3>
<p>Finally, we can create a shiny app. This example app can be found <a href="https://bioinf.stemcells.cam.ac.uk/shiny/bulkAnalyseR/GEO/" class="external-link">here</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">bulkAnalyseR</span><span class="fu">::</span><span class="fu"><a href="../reference/generateShinyApp.html">generateShinyApp</a></span><span class="op">(</span></span>
<span>  shiny.dir <span class="op">=</span> <span class="st">"shiny_GEO"</span>,</span>
<span>  app.title <span class="op">=</span> <span class="st">"Shiny app for visualisation of GEO data"</span>,</span>
<span>  modality <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  expression.matrix <span class="op">=</span> <span class="va">exp.proc</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">meta</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"hsapiens"</span>,</span>
<span>  org.db <span class="op">=</span> <span class="st">"org.Hs.eg.db"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">##&gt; R version 4.1.1 (2021-08-10)</span></span>
<span><span class="co">##&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">##&gt; Running under: Windows 10 x64 (build 22000)</span></span>
<span><span class="co">##&gt; </span></span>
<span><span class="co">##&gt; Matrix products: default</span></span>
<span><span class="co">##&gt; </span></span>
<span><span class="co">##&gt; locale:</span></span>
<span><span class="co">##&gt; [1] LC_COLLATE=English_United Kingdom.1252 </span></span>
<span><span class="co">##&gt; [2] LC_CTYPE=English_United Kingdom.1252   </span></span>
<span><span class="co">##&gt; [3] LC_MONETARY=English_United Kingdom.1252</span></span>
<span><span class="co">##&gt; [4] LC_NUMERIC=C                           </span></span>
<span><span class="co">##&gt; [5] LC_TIME=English_United Kingdom.1252    </span></span>
<span><span class="co">##&gt; </span></span>
<span><span class="co">##&gt; attached base packages:</span></span>
<span><span class="co">##&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">##&gt; </span></span>
<span><span class="co">##&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">##&gt;  [1] Rcpp_1.0.8.3          lattice_0.20-44       assertthat_0.2.1     </span></span>
<span><span class="co">##&gt;  [4] rprojroot_2.0.3       digest_0.6.29         foreach_1.5.2        </span></span>
<span><span class="co">##&gt;  [7] utf8_1.2.2            mime_0.12             R6_2.5.1             </span></span>
<span><span class="co">##&gt; [10] evaluate_0.16         highr_0.9             ggplot2_3.3.5        </span></span>
<span><span class="co">##&gt; [13] pillar_1.8.1          rlang_1.0.6           rstudioapi_0.14      </span></span>
<span><span class="co">##&gt; [16] jquerylib_0.1.4       Matrix_1.5-1          preprocessCore_1.54.0</span></span>
<span><span class="co">##&gt; [19] rmarkdown_2.16        pkgdown_2.0.6         textshaping_0.3.6    </span></span>
<span><span class="co">##&gt; [22] desc_1.4.2            labeling_0.4.2        splines_4.1.1        </span></span>
<span><span class="co">##&gt; [25] stringr_1.4.1         munsell_0.5.0         philentropy_0.6.0    </span></span>
<span><span class="co">##&gt; [28] shiny_1.7.1           compiler_4.1.1        httpuv_1.6.3         </span></span>
<span><span class="co">##&gt; [31] xfun_0.29             pkgconfig_2.0.3       systemfonts_1.0.4    </span></span>
<span><span class="co">##&gt; [34] mgcv_1.8-36           htmltools_0.5.2       tidyselect_1.1.2     </span></span>
<span><span class="co">##&gt; [37] tibble_3.1.6          codetools_0.2-18      fansi_1.0.2          </span></span>
<span><span class="co">##&gt; [40] dplyr_1.0.8           later_1.3.0           noisyr_1.0.0         </span></span>
<span><span class="co">##&gt; [43] grid_4.1.1            nlme_3.1-152          jsonlite_1.8.0       </span></span>
<span><span class="co">##&gt; [46] xtable_1.8-4          gtable_0.3.1          lifecycle_1.0.2      </span></span>
<span><span class="co">##&gt; [49] DBI_1.1.3             magrittr_2.0.2        scales_1.2.1         </span></span>
<span><span class="co">##&gt; [52] cli_3.4.1             stringi_1.7.6         cachem_1.0.6         </span></span>
<span><span class="co">##&gt; [55] farver_2.1.0          fs_1.5.2              promises_1.2.0.1     </span></span>
<span><span class="co">##&gt; [58] bslib_0.4.0           ellipsis_0.3.2        ragg_1.2.2           </span></span>
<span><span class="co">##&gt; [61] generics_0.1.3        vctrs_0.3.8           iterators_1.0.14     </span></span>
<span><span class="co">##&gt; [64] tools_4.1.1           glue_1.6.2            bulkAnalyseR_1.0.0   </span></span>
<span><span class="co">##&gt; [67] purrr_0.3.4           fastmap_1.1.0         yaml_2.3.5           </span></span>
<span><span class="co">##&gt; [70] colorspace_2.0-3      memoise_2.0.1         knitr_1.40           </span></span>
<span><span class="co">##&gt; [73] sass_0.4.1</span></span></code></pre></div>
</div>
</div>
  <aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="copyright">
  <p></p>
<p>Developed by Ilias Moutsopoulos, Eleanor Williams, Irina Mohorianu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

    </footer></main>
</div>

  

  

  </div>
</body>
</html>
