<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light"><script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/5/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Function reference â€¢ bulkAnalyseR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/5/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/5/darkly/bootstrap.css" media="(prefers-color-scheme: dark)"><!-- preferably CSS --><link rel="stylesheet" href="../preferably.css"><link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)"><script src="../darkswitch.js"></script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">bulkAnalyseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/bulkAnalyseR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GEO_App.html">A beginner's guide to creating a bulkAnalyseR app from a GEO dataset</a>
    <a class="dropdown-item" href="../articles/IntegratingMultiomics.html">bulkAnalyseR: Integrating multi-omics or external data</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Core-Bioinformatics/bulkAnalyseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>

        
        <li>
          <a class="external-link nav-link" id="css-toggle-btn" aria-label="github">
            <span class="fas fa fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        
        
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="calculate_condition_mean_sd_per_gene.html">calculate_condition_mean_sd_per_gene()</a></code> 
        </dt>
        <dd>Calculate statistics for each gene of an expression matrix given a grouping</dd>
      </dl><dl><dt>
          
          <code><a href="crossPanel.html">crossPanelUI()</a></code> <code><a href="crossPanel.html">crossPanelServer()</a></code> 
        </dt>
        <dd>Generate the cross plot panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="cross_plot.html">cross_plot()</a></code> 
        </dt>
        <dd>Create a cross plot comparing differential expression (DE) results</dd>
      </dl><dl><dt>
          
          <code><a href="DEanalysis.html">DEanalysis_edger()</a></code> <code><a href="DEanalysis.html">DEanalysis_deseq2()</a></code> 
        </dt>
        <dd>Perform differential expression (DE) analysis on an expression matrix</dd>
      </dl><dl><dt>
          
          <code><a href="DEpanel.html">DEpanelUI()</a></code> <code><a href="DEpanel.html">DEpanelServer()</a></code> 
        </dt>
        <dd>Generate the DE panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="DEplotPanel.html">DEplotPanelUI()</a></code> <code><a href="DEplotPanel.html">DEplotPanelServer()</a></code> 
        </dt>
        <dd>Generate the DE plot plot panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="DEsummaryPanel.html">DEsummaryPanelUI()</a></code> <code><a href="DEsummaryPanel.html">DEsummaryPanelServer()</a></code> 
        </dt>
        <dd>Generate the DE summary panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="determine_uds.html">determine_uds()</a></code> 
        </dt>
        <dd>Determine the pattern between two intervals</dd>
      </dl><dl><dt>
          
          <code><a href="enrichmentPanel.html">enrichmentPanelUI()</a></code> <code><a href="enrichmentPanel.html">enrichmentPanelServer()</a></code> 
        </dt>
        <dd>Generate the enrichment panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="expression_heatmap.html">expression_heatmap()</a></code> 
        </dt>
        <dd>Create heatmap of an expression matrix</dd>
      </dl><dl><dt>
          
          <code><a href="find_regulators_with_recurring_edges.html">find_regulators_with_recurring_edges()</a></code> 
        </dt>
        <dd>Find recurring regulators</dd>
      </dl><dl><dt>
          
          <code><a href="generateShinyApp.html">generateShinyApp()</a></code> 
        </dt>
        <dd>Generate all files required for an autonomous shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="get_link_list_rename.html">get_link_list_rename()</a></code> 
        </dt>
        <dd>Convert the adjacency matrix to network links</dd>
      </dl><dl><dt>
          
          <code><a href="GRNCisPanel.html">GRNCisPanelUI()</a></code> <code><a href="GRNCisPanel.html">GRNCisPanelServer()</a></code> 
        </dt>
        <dd>Generate the GRN cis integration panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="GRNCustomPanel.html">GRNCustomPanelUI()</a></code> <code><a href="GRNCustomPanel.html">GRNCustomPanelServer()</a></code> 
        </dt>
        <dd>Generate the GRN custom integration panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="GRNpanel.html">GRNpanelUI()</a></code> <code><a href="GRNpanel.html">GRNpanelServer()</a></code> 
        </dt>
        <dd>Generate the GRN panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="GRNTransPanel.html">GRNTransPanelUI()</a></code> <code><a href="GRNTransPanel.html">GRNTransPanelServer()</a></code> 
        </dt>
        <dd>Generate the GRN trans integration panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="infer_GRN.html">infer_GRN()</a></code> 
        </dt>
        <dd>Perform GRN inference</dd>
      </dl><dl><dt>
          
          <code><a href="jaccard_heatmap.html">jaccard_heatmap()</a></code> 
        </dt>
        <dd>Create a heatmap of the Jaccard similarity index (JSI) between samples of
an experiment</dd>
      </dl><dl><dt>
          
          <code><a href="jaccard_index.html">jaccard_index()</a></code> 
        </dt>
        <dd>Calculate the Jaccard similarity index (JSI) between two vectors</dd>
      </dl><dl><dt>
          
          <code><a href="landingPanel.html">landingPanelUI()</a></code> <code><a href="landingPanel.html">landingPanelServer()</a></code> 
        </dt>
        <dd>Generate the landing page panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="make_heatmap_matrix.html">make_heatmap_matrix()</a></code> 
        </dt>
        <dd>Create a matrix of the average expression of each gene in each condition</dd>
      </dl><dl><dt>
          
          <code><a href="make_pattern_matrix.html">make_pattern_matrix()</a></code> 
        </dt>
        <dd>Create a matrix of the patterns between conditions</dd>
      </dl><dl><dt>
          
          <code><a href="ma_plot.html">ma_plot()</a></code> <code><a href="ma_plot.html">ma_enhance()</a></code> 
        </dt>
        <dd>Create an MA plot visualising differential expression (DE) results</dd>
      </dl><dl><dt>
          
          <code><a href="modalityPanel.html">modalityPanelUI()</a></code> <code><a href="modalityPanel.html">modalityPanelServer()</a></code> 
        </dt>
        <dd>Generate an app panel for a modality</dd>
      </dl><dl><dt>
          
          <code><a href="noisyr_counts_with_plot.html">noisyr_counts_with_plot()</a></code> 
        </dt>
        <dd>Apply a modified noisyR counts pipeline printing a plot</dd>
      </dl><dl><dt>
          
          <code><a href="patternPanel.html">patternPanelUI()</a></code> <code><a href="patternPanel.html">patternPanelServer()</a></code> 
        </dt>
        <dd>Generate the expression patterns panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="plot_GRN.html">plot_GRN()</a></code> 
        </dt>
        <dd>Plot a GRN</dd>
      </dl><dl><dt>
          
          <code><a href="plot_line_pattern.html">plot_line_pattern()</a></code> 
        </dt>
        <dd>Create a line plot of average expression across conditions</dd>
      </dl><dl><dt>
          
          <code><a href="plot_pca.html">plot_pca()</a></code> 
        </dt>
        <dd>Create a principal component analysis (PCA) plot the samples of an experiment</dd>
      </dl><dl><dt>
          
          <code><a href="plot_upset.html">plot_upset()</a></code> 
        </dt>
        <dd>Visualise the overlap of edges between different networks</dd>
      </dl><dl><dt>
          
          <code><a href="preprocessExpressionMatrix.html">preprocessExpressionMatrix()</a></code> 
        </dt>
        <dd>Pre-process the expression matrix before generating the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="preprocess_miRTarBase.html">preprocess_miRTarBase()</a></code> 
        </dt>
        <dd>Creates a comparison table for miRTarBase to be used for custom integration</dd>
      </dl><dl><dt>
          
          <code><a href="QCpanel.html">QCpanelUI()</a></code> <code><a href="QCpanel.html">QCpanelServer()</a></code> 
        </dt>
        <dd>Generate the QC panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="rescale_matrix.html">rescale_matrix()</a></code> 
        </dt>
        <dd>Rescale a matrix</dd>
      </dl><dl><dt>
          
          <code><a href="sampleSelectPanel.html">sampleSelectPanelUI()</a></code> <code><a href="sampleSelectPanel.html">sampleSelectPanelServer()</a></code> 
        </dt>
        <dd>Generate the sample select panel of the shiny app</dd>
      </dl><dl><dt>
          
          <code><a href="volcano_plot.html">volcano_plot()</a></code> <code><a href="volcano_plot.html">volcano_enhance()</a></code> 
        </dt>
        <dd>Create a volcano plot visualising differential expression (DE) results</dd>
      </dl></div>
  </main></div>


    <footer><div class="copyright">
  <p></p><p>Developed by Ilias Moutsopoulos, Eleanor Williams, Irina Mohorianu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

    </footer></div>

  

  

  </body></html>

