<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function denoises the expression matrix using the noisyR package
and then normalises it. It is recommended to use this function before using
generateShinyApp."><!-- Inform modern browsers that this page supports both dark and light color schemes,
  and the page author prefers light. --><meta name="color-scheme" content="dark light"><script>
  // If `prefers-color-scheme` is not supported, fall back to light mode.
  // i.e. In this case, inject the `light` CSS before the others, with
  // no media filter so that it will be downloaded with highest priority.
  if (window.matchMedia("(prefers-color-scheme: dark)").media === "not all") {
    document.documentElement.style.display = "none";
    document.head.insertAdjacentHTML(
      "beforeend",
      "<link id=\"css\" rel=\"stylesheet\" href=\"https://bootswatch.com/5/flatly/bootstrap.css\" onload=\"document.documentElement.style.display = ''\">"
    );
  }
</script><title>Pre-process the expression matrix before generating the shiny app — preprocessExpressionMatrix • bulkAnalyseR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pre-process the expression matrix before generating the shiny app — preprocessExpressionMatrix"><meta property="og:description" content="This function denoises the expression matrix using the noisyR package
and then normalises it. It is recommended to use this function before using
generateShinyApp."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Flatly Theme - Light  --><link id="css-light" rel="stylesheet" href="https://bootswatch.com/5/flatly/bootstrap.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><!-- Darkly Theme - Dark --><link id="css-dark" rel="stylesheet" href="https://bootswatch.com/5/darkly/bootstrap.css" media="(prefers-color-scheme: dark)"><!-- preferably CSS --><link rel="stylesheet" href="../preferably.css"><link id="css-code-light" rel="stylesheet" href="../code-color-scheme-light.css" media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"><link id="css-code-dark" rel="stylesheet" href="../code-color-scheme-dark.css" media="(prefers-color-scheme: dark)"><script src="../darkswitch.js"></script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">bulkAnalyseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/bulkAnalyseR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/IntegratingMultiomics.html">bulkAnalyseR: Integrating multi-omics or external data</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/Core-Bioinformatics/bulkAnalyseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>

        
        <li>
          <a class="external-link nav-link" id="css-toggle-btn" aria-label="github">
            <span class="fas fa fas fa-adjust fa-lg"></span>
          </a>
        </li>
        
        
        
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Pre-process the expression matrix before generating the shiny app</h1>
      <small class="dont-index">Source: <a href="https://github.com/Core-Bioinformatics/bulkAnalyseR/blob/HEAD/R/preprocessExpressionMatrix.R" class="external-link"><code>R/preprocessExpressionMatrix.R</code></a></small>
      <div class="d-none name"><code>preprocessExpressionMatrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function denoises the expression matrix using the noisyR package
and then normalises it. It is recommended to use this function before using
<code><a href="generateShinyApp.html">generateShinyApp</a></code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">preprocessExpressionMatrix</span><span class="op">(</span>
  <span class="va">expression.matrix</span>,
  denoise <span class="op">=</span> <span class="cn">TRUE</span>,
  output.plot <span class="op">=</span> <span class="cn">FALSE</span>,
  normalisation.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"quantile"</span>, <span class="st">"rpm"</span>, <span class="st">"tmm"</span>, <span class="st">"deseq2"</span><span class="op">)</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>expression.matrix</dt>
<dd><p>the expression matrix; rows correspond to genes and
columns correspond to samples</p></dd>
<dt>denoise</dt>
<dd><p>whether to use noisyR to denoise the expression matrix;
proceeding without denoising data is not recommended</p></dd>
<dt>output.plot</dt>
<dd><p>whether to create an expression-similarity plot for the
noise analysis (printed to the console); default is FALSE</p></dd>
<dt>normalisation.method</dt>
<dd><p>the normalisation method to be used; default is
quantile; any unrecognised input will result in no normalisation being
applied, but proceeding with un-normalised data is not recommended;
currently supported normalisation methods are:</p><dl><dt>quantile</dt>
<dd><p>Quantile normalisation using the <code>normalize.quantiles</code>
function from the <code>preprocessCore</code> package</p></dd>

<dt>rpm</dt>
<dd><p>A version of RPM (reads per million) normalisation, where each
sample is scaled by the median expression in the sample divided by the
total number of reads in that sample</p></dd>

<dt>tmm</dt>
<dd><p>Trimmed Mean of M values normalisation using the
<code>calcNormFactors</code> function from the <code>edgeR</code> package</p></dd>

<dt>deseq2</dt>
<dd><p>Size factor normalisation using the
<code>estimateSizeFactorsForMatrix</code> function from the <code>DESeq2</code> package</p></dd>


</dl></dd>
<dt>...</dt>
<dd><p>optional arguments passed on to <code><a href="https://rdrr.io/pkg/noisyr/man/noisyr_counts.html" class="external-link">noisyr::noisyr_counts()</a></code></p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The denoised, normalised expression matrix; some rows (genes)
may have been removed by noisyR.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">expression.matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span></span>
<span class="r-in">  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"expression_matrix.csv"</span>, package <span class="op">=</span> <span class="st">"bulkAnalyseR"</span><span class="op">)</span>, </span>
<span class="r-in">  row.names <span class="op">=</span> <span class="fl">1</span></span>
<span class="r-in"><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">500</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span class="r-in"><span class="va">expression.matrix.preproc</span> <span class="op">&lt;-</span> <span class="fu">preprocessExpressionMatrix</span><span class="op">(</span><span class="va">expression.matrix</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt;&gt; noisyR counts approach pipeline &lt;&lt;&lt;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The input matrix has 500 rows and 4 cols</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     number of genes: 500</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     number of samples: 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating the number of elements per window</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the number of elements per window is 50</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the step size is 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     the selected similarity metric is correlation_pearson</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Working with sample 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Working with sample 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Working with sample 3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   Working with sample 4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating noise thresholds for 4 samples...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     similarity.threshold = 0.25</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     method.chosen = Boxplot-IQR</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Denoising expression matrix...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     removing noisy genes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     adjusting matrix</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt;&gt; Done! &lt;&lt;&lt;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Performing quantile normalisation...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-plt img"><img src="preprocessExpressionMatrix-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="copyright">
  <p></p><p>Developed by Ilias Moutsopoulos, Eleanor Williams, Irina Mohorianu.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
  <p class="preferably">Using <a href="https://preferably.amirmasoudabdol.name/?source=footer" class="external-link">preferably</a> template.</p>
</div>

    </footer></div>

  

  

  </body></html>

